test/adapter/json_api/parse_test.rb

```diff
@@ -0,0 +1,67 @@
+require 'test_helper'
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        module Deserialization
+          class ParseTest < Minitest::Test
+            def setup
```

Comment:
Added an integration test. However, after giving it some thoughts, I don't see the relevance of a 'reversible' test. What we are doing here is precisely transforming a JSON API-compliant document into an `ActiveRecord::Attributes`-compliant hash, so I'm not sure what introducing a dependency to the serialization would add.